docusaurus/src/remark/docsHeaderDecoration.js

```diff
@@ -0,0 +1,135 @@
+const fetch = require("node-fetch");
+const visit = require("unist-util-visit");
+
+const registry_url =
+  "https://connectors.airbyte.com/files/generated_reports/connector_registry_report.json";
+let registry = {};
+let loading = false;
+
+const plugin = () => {
+  const transformer = async (ast, vfile) => {
+    if (!isDocsPage(vfile)) return;
+
+    await fetchCatalog();
+
+    const pathParts = vfile.path.split("/");
+    const connectorName = pathParts.pop().split(".")[0];
+    const connectorType = pathParts.pop();
+    const dockerRepository = `airbyte/${connectorType.replace(
+      /s$/,
+      ""
+    )}-${connectorName}`;
+    const registryEntry = registry.find(
+      (r) => r.dockerRepository_oss === dockerRepository
+    );
+
+    if (!registryEntry) return;
+
+    visit(ast, "heading", (node) => {
+      let headerFound = false;
```

Comment:
This does not work the way you expect. Since this is a block level variable this will always be false in the `if` below. You likely just want to pull the type declaration outside of the visit function. Should likely be easy to test by just adding another H1 in a doc.